### DISCUSSION TITLE
[11주차] 네트워크 보안 8.5~9

### DISCUSSION THREAD

**Discussion by Hanjaemo** [Dijkstra-Employ/Network-Study#18] [Q&A]
11주차 네트워크 보안 8.5~9에 해당하는 질의응답입니다.

**Comment by Starlight258** [comment]
# Q1
전자우편의 보안은 어떻게 달성되나요?

# Q2
SSL 핸드쉐이크 과정에 대해 설명해주세요.

**Reply by H-sooyeon** [reply]
> Q1 전자우편의 보안은 어떻게 달성되나요?

애플리케이션 계층에 속하는 SSL/TLS 보안 계층에서 보안을 제공한다.
TLS는 기밀성, 데이터 무결성, 서버 인증과 클라이언트 인증을 통해 TCP를 향상함으로써 여러 보안 문제를 해결한다.

<br />

> Q2 SSL 핸드쉐이크 과정에 대해 설명해주세요.

SSL은 TLS가 나오기 이전에 사용되던 보안 프로토콜이며, 보안상에 문제가 있어 더는 사용되지 않는다.
대신, TLS를 사용하고 있다.
TLS는 다음과 같은 과정을 거쳐 핸드셰이크를 진행한다.

1. 송신자는 수신자에게 Hello 메시지를 보낸다.
2. 수신자는 수신자의 공개키를 담은 인증서로 응답한다.
3. 송신자는 MS(Master secret)을 생성하고, 이를 수신자의 공개키로 암호화하여 EMS(Encrypted Master Secret)을 만든다.
4. 이 EMS가 수신자에게 전송되고, 수신자는 자신의 개인키를 가지고 EMS를 복호화함으로써 MS를 얻는다.
5. 송신자와 수신자는 MS로부터 각자 4개의 키를 생성한다. (각각의 세션 암호화 키, 세션 HMAC 키)
6. 2개의 암호화 키는 데이터를 암호화하는 데 쓰이고, 2개의 HMAC 키는 데이터 무결성을 호가인하는 데 사용된다.

**Comment by H-sooyeon** [comment]
## Q1
네트워크 계층의 보안 포로토콜로 IPsec이 있습니다.
이는 일반적으로 VPN과 함께 구현되는데, 어떤 과정을 거쳐 보안을 제공할까요?

## Q2
방화벽의 종류와 각 특징을 간단히 설명해주세요.
